man/Tss2_Tcti_Libtpms_Init.3.in

```diff
@@ -0,0 +1,113 @@
+.\" Process this file with
+.\" groff -man -Tascii foo.1
+.\"
+.TH Tss2_Tcti_Libtpms_Init 3 "FEBRUARY 2021" "TPM2 Software Stack"
+.SH NAME
+Tss2_Tcti_Libtpms_Init \- Initialization function for the libtpms TPM simulator TCTI library.
+.SH SYNOPSIS
+.B #include <tcti/tcti_libtpms.h>
+.sp
+.BI "TSS2_RC Tss2_Tcti_Libtpms_Init (TSS2_TCTI_CONTEXT " "*tctiContext" ", size_t " "*contextSize" ", const char " "*conf" ");"
+.sp
+The
+.BR  Tss2_Tcti_Libtpms_Init ()
+function initializes a TCTI context used to communicate with the libtpms TPM2
+simulator.
+.SH DESCRIPTION
+.BR Tss2_Tcti_Libtpms_Init ()
+attempts to dynamically load the libtpms library and initialize a caller allocated
+.I tcti_context
+of size
+.I size
+\&. Since the
+.I tcti_context
+must be a caller allocated buffer, the caller needs to know the buffer size
+required by the TCTI library. The minimum size of this context can be
+discovered by providing
+.BR NULL
+for the
+.I tcti_context
+and a non-
+.BR NULL
+.I size
+parameter. The initialization function will then populate the
+.I size
+parameter with the minimum size of the
+.I tcti_context
+buffer. The caller must then allocate a buffer of this size (or larger) and
+call
+.B Tss2_Tcti_Libtpms_Init ()
+again providing the newly allocated
+.I tcti_context
+and the size of this context in the
+.I size
+parameter. This pattern is common to all TCTI initialization functions. We
+provide an example of this pattern using the
+.BR Tss2_Tcti_Libtpms_Init ()
+function in the section titled
+.B EXAMPLE.
+.sp
+The
+.I conf
+parameter is currently unused.
```

Comment:
can we use this for `--tpmstate dir` setting ?

Reply:
That was exactly what I was thinking about. We could register callbacks to dump state into a file. This is a bit of work, though, because you can either register no callback or all of them... which you then have to implement. This is a topic for another day and another PR ;)

Reply:
Oh, ok. Maybe you could add this info to the doc, that the state is not persistenced.
I assumed it would just write to $PWD.

Reply:
I reworked the PR, including having a state file.